components/msg/compositor_msg.rs

```diff
@@ -36,35 +36,62 @@ impl FrameTreeId {
 }
 
 #[derive(Clone, PartialEq, Eq, Copy, Hash, Deserialize, Serialize, HeapSizeOf)]
-pub struct LayerId(
-    /// A base layer ID, currently derived from DOM element pointer address.
-    pub usize,
-
-    /// FIXME(#2010, pcwalton): A marker for overflow scroll layers.
-    pub u32,
+pub enum FragmentPart {
```

Comment:
Maybe something like `LayerType` or `LayerKind`? The reason is that we'll eventually want canvas and WebGL layers (maybe image layers too).